pkg/blockstorage/awsebs/awsebs.go

```diff
@@ -456,6 +456,9 @@ func (s *EbsStorage) VolumeCreateFromSnapshot(ctx context.Context, snapshot bloc
 
 	volID, err := createVolume(ctx, s.Ec2Cli, cvi, ktags.GetTags(tags))
 	if err != nil {
+		if isVolNotFoundErr(err) {
+			return nil, errors.Wrap(err, "This may indicate insufficient permissions for KMS keys.")
+		}
```

Comment:
@bathina2 

I took a look  the isVolNotFoundErr

```

func isVolNotFoundErr(err error) bool {
	return isError(err, volumeNotFoundCode)
}
```

```
func isError(err error, code string) bool {
	awsErr, ok := err.(awserr.Error)
	return ok && awsErr.Code() == code
}
```

This the AWS error interface:

```
type Error interface {
	// Satisfy the generic error interface.
	error

	// Returns the short phrase depicting the classification of the error.
	Code() string

	// Returns the error details message.
	Message() string

	// Returns the original error if one was set.  Nil is returned if not set.
	OrigErr() error
}
```

Can you check if there is anything useful returned using Message or OrigErr ? Can we parse that to get the exact reason for the failure during createVolume ?

Reply:
comment in awserr/error.go :   

```
// Example:
//
//     output, err := s3manage.Upload(svc, input, opts)
//     if err != nil {
//         if awsErr, ok := err.(awserr.Error); ok {
//             // Get error details
//             log.Println("Error:", awsErr.Code(), awsErr.Message())
//
//             // Prints out full error message, including original error if there was one.
//             log.Println("Error:", awsErr.Error())
//
//             // Get original error
//             if origErr := awsErr.OrigErr(); origErr != nil {
//                 // operate on original error.
//             }
//         } else {
//             fmt.Println(err.Error())
//         }
//     }
//
```

Reply:
`"InvalidVolume.NotFound: The volume 'vol-08d83a8945eed3580' does not exist.\\n\\tstatus code: 400, request id: 6b647f82-bcd3-4957-8061-32070bba10dd\"`

Reply:
@onkarbhat refer to this ticket - https://kasten.atlassian.net/browse/K10-2613

Reply:
Thats the full error. Doesn't look like theres any original error set.

Reply:
`

func (r requestError) Error() string {
--
  |   | extra := fmt.Sprintf("status code: %d, request id: [%s]",
  |   | r.statusCode, r.requestID)
  |   | return SprintError(r.Code(), r.Message(), extra, r.OrigErr())
  |   | }

`

Reply:
The error output contains the OrigErr.

Reply:
Thanks for checking @bathina2 . LGTM